code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/identity/TokenResourceManager.java

```diff
@@ -115,4 +133,63 @@ public void setTokenCreators(List<TokenCreator> tokenCreators) {
         this.tokenCreators = tokenCreators;
     }
 
+    @Override
+    protected Object deleteInternal(String type, String id, Object obj, ApiRequest request) {
+        if (!StringUtils.equals(AbstractTokenUtil.TOKEN, request.getType())) {
+            return null;
+        }
+        return deleteToken(obj, request);
+    }
+
+    protected Object deleteToken(Object obj, ApiRequest request) {
+        Token token = new Token();
+        String jwt = "";
+
+        token = listToken();
+        jwt = token.getJwt();
+
+        if(StringUtils.isBlank(jwt)) {
+            throw new ClientVisibleException(ResponseCodes.INTERNAL_SERVER_ERROR,
+                    "JWTNotProvided", "Request does not contain JWT cookie", null);
+        }
+
+        request.setResponseCode(ResponseCodes.NO_CONTENT);
+        HttpServletResponse response = request.getServletContext().getResponse();
+        HttpServletRequest httpRequest = request.getServletContext().getRequest();
+        List<Cookie> cookies = getAllCookies(httpRequest, "token");
+        if (cookies != null) {
+            if (cookies.size() == 2) {
+                cookies.get(0).setPath("/");
+                cookies.get(1).setPath("/v2-beta/token");
+            }
+
+            for (Cookie cookie : cookies) {
+                cookie.setValue("");
+                cookie.setMaxAge(0);
```

Comment:
I spoke with @vincent99 and unfortunately, max-age is not supported by internet explorer. So, you need to also set expires. The java cookie api doesnt have a setter for that, so you need to switch to adding a cookie via the set-cookie header. The value can just be a constant string that has a blank value, expires date in the past, and whatever other fields are required.

Also, I think you just need to do it once, you dont need to get all the cookies and loop through them.